src/js/components/Tip/Tip.js

```diff
@@ -0,0 +1,88 @@
+import React, {
+  Children,
+  cloneElement,
+  forwardRef,
+  useContext,
+  useState,
+} from 'react';
+import { ThemeContext } from 'styled-components';
+
+import { Box } from '../Box';
+import { Drop } from '../Drop';
+import { useForwardedRef } from '../../utils/refs';
+
+const Tip = forwardRef(({ children, content, dropProps }, tipRef) => {
+  const theme = useContext(ThemeContext);
+  const [over, setOver] = useState(false);
+
+  const componentRef = useForwardedRef(tipRef);
+
+  // In cases the child is a primitive
+  const wrapInvalidElement = () => {
```

Comment:
Maybe call this `wrapPrimitiveElement`?

Reply:
I'm not sure if there is 1:1 correlation between primitives and invalidReactElements. Semantically, the function is wrapping invalid React elements.